flow360/component/volume_mesh.py

```diff
@@ -452,8 +461,30 @@ def _submit_upload_mesh(self, progress_callback=None):
         info = VolumeMeshMeta(**resp)
         self._id = info.id
         mesh = VolumeMesh(self.id)
-        mesh.upload_file(remote_file_name, self.file_name, progress_callback=progress_callback)
-        mesh._complete_upload(remote_file_name)
+
+        # parallel compress and upload
+        if (
+            original_compression == CompressionFormat.NONE
+            and self.compress_method == CompressionFormat.BZ2
+        ):
+            upload_id = mesh.create_multipart_upload(remote_file_name)
+            compress_and_upload_chunks(self.file_name, upload_id, mesh, remote_file_name)
+        else:
+            if original_compression == CompressionFormat.NONE:
```

Comment:
check explicitly for ZST compress method to have option to turn off compression and upload file directly